openfisca_uk_data/datasets/frs/frs_was_imputation.py

```diff
@@ -0,0 +1,191 @@
+from openfisca_uk_data.utils import dataset, UK
+from openfisca_uk_data.datasets.frs.frs import FRS
+import pandas as pd
+import microdf as mdf
+import sklearn
+import synthimpute as si
+import h5py
+import numpy as np
+
+
+@dataset
+class FRS_WAS_Imputation:
+    name = "frs_was_imp"
+    model = UK
+
+    def generate(was_file: str, year: int):
+        was_df = process_was(was_file)
+        pred_land = impute_land(was_df, year)
+        frs_was = h5py.File(FRS_WAS_Imputation.file(year), mode="w")
+        frs = FRS.load(year)
+        for variable in tuple(frs.keys()):
+            frs_was[variable] = np.array(frs[variable][...])
+        frs_was["land_value"] = pred_land.values
+        frs_was.close()
+        frs.close()
+
+
+def impute_land(was_df: pd.DataFrame, year: int) -> pd.Series:
+    """Impute land by fitting a random forest model.
+
+    Args:
+            was_df (pd.DataFrame): The WAS dataset.
+            year (int): The year of simulation.
+
+    Returns:
+            pd.Series: The predicted land values.
+    """
+    from openfisca_uk import Microsimulation
+
+    sim = Microsimulation(dataset=FRS, year=year)
+
+    was = pd.read_csv("~/was.csv")
+
+    TRAIN_COLS = [
+        "gross_income",
+        "num_adults",
+        "num_children",
+        "pension_income",
+        "employment_income",
+        "self_employment_income",
+        "investment_income",
+        "num_bedrooms",
+        "council_tax",
+        "is_renting",
+        # "region"  # TODO: add region mapping from WAS int.
+    ]
+
+    IMPUTE_COLS = [
+        "est_land",  # Total wealth.
+    ]
+
+    # TODO: Test imputation quantiles on a hold out sample.
```

Comment:
we can remove this and report model quality on a holdout set somewhere else in the future

Reply:
OK yep, leaving in until then